lighthouse-core/gather/gatherers/css-usage.js

```diff
@@ -93,25 +116,26 @@ class CSSUsage extends FRGatherer {
       await this.stopCSSUsageTracking(context);
     }
 
-    // Fetch style sheet content in parallel.
-    const promises = this._stylesheets.map(sheet => {
-      const styleSheetId = sheet.header.styleSheetId;
-      return session.sendCommand('CSS.getStyleSheetText', {styleSheetId}).then(content => {
-        return {
-          header: sheet.header,
-          content: content.text,
-        };
-      });
-    });
-    const styleSheetInfo = await Promise.all(promises);
+    /** @type {Map<string, LH.Artifacts.CSSStyleSheetInfo>} */
+    const dedupedStylesheets = new Map();
+    const sheets = await Promise.all(this._sheetPromises.values());
+    const sheetIds = Array.from(this._sheetPromises.keys());
+
+    for (let i = 0; i < sheets.length; ++i) {
+      const sheet = sheets[i];
+      const sheetId = sheetIds[i];
+
+      if (sheet instanceof Error) {
+        if (this._removedSheets.has(sheetId)) continue;
+        throw sheet;
+      }
```

Comment:
this error handling looks new, can you say more about why it's needed now?

Reply:
This was included to pass one of the existing unit tests where removed stylesheets are ignored. 

However, that logic was also included to prevent errors when fetching the content of removed stylesheets. With this PR, we should be able to fetch the content of removed stylesheets.

Reply:
As for handling the errors, I don't think throwing the error in the event handler will surface that error in the report. Throwing the error here will surface it in the report.

Reply:
It also invalidates all the other valid usages found. Can this be a warning instead?